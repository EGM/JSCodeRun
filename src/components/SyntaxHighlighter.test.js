import React, { Component } from 'react';
import renderer from 'react-test-renderer';
import SyntaxHighlighter from './SyntaxHighlighter';

describe('SyntaxHighlighter', () => {
  it('should colorize basic keywords', () => {
    const rendered = renderer.create(<SyntaxHighlighter text="var something" />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('should colorize function-type keywords', () => {
    const rendered = renderer.create(<SyntaxHighlighter text="function something" />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('should colorize advanced-type keywords', () => {
    const rendered = renderer.create(<SyntaxHighlighter text="class something" />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('should colorize strings', () => {
    const rendered = renderer
      .create(<SyntaxHighlighter text={'`one string` \'two string\' "three string"'} />)
      .toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('should colorize numbers', () => {
    const rendered = renderer.create(<SyntaxHighlighter text="number 3" />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('should colorize comments', () => {
    const rendered = renderer.create(<SyntaxHighlighter text="something //with comments\n" />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('should keep other words default color', () => {
    const rendered = renderer.create(<SyntaxHighlighter text="something" />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
