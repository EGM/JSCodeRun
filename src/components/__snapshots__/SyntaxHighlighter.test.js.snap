// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SyntaxHighlighter should colorize advanced-type keywords 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#eee",
      }
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#71d5c3",
      }
    }
  >
    class
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#eee",
      }
    }
  >
     something
  </Text>
</Text>
`;

exports[`SyntaxHighlighter should colorize basic keywords 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#eee",
      }
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#769aa5",
      }
    }
  >
    var
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#eee",
      }
    }
  >
     something
  </Text>
</Text>
`;

exports[`SyntaxHighlighter should colorize comments 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#eee",
      }
    }
  >
    something //
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#71d5c3",
      }
    }
  >
    with
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#eee",
      }
    }
  >
     comments\\n
  </Text>
</Text>
`;

exports[`SyntaxHighlighter should colorize function-type keywords 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#eee",
      }
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#9876aa",
      }
    }
  >
    function
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#eee",
      }
    }
  >
     something
  </Text>
</Text>
`;

exports[`SyntaxHighlighter should colorize numbers 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#eee",
      }
    }
  >
    number 
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#a5c25c",
      }
    }
  >
    3
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#eee",
      }
    }
  >
    
  </Text>
</Text>
`;

exports[`SyntaxHighlighter should colorize strings 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#eee",
      }
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#cc7832",
      }
    }
  >
    \`one string\`
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#eee",
      }
    }
  >
     
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#cc7832",
      }
    }
  >
    'two string'
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#eee",
      }
    }
  >
     
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#cc7832",
      }
    }
  >
    "three string"
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#eee",
      }
    }
  >
    
  </Text>
</Text>
`;

exports[`SyntaxHighlighter should keep other words default color 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#eee",
      }
    }
  >
    something
  </Text>
</Text>
`;
